File: redwood-yoga-java/src/main/java/io/github/orioncraftmc/meditate/internal/YGCollectFlexItemsRowValues.java
package io.github.orioncraftmc.meditate.internal;

import java.util.ArrayList;
import org.jetbrains.annotations.NotNull;

public class YGCollectFlexItemsRowValues //Type originates from: Utils.h
{
    public int itemsOnLine;
    public float sizeConsumedOnCurrentLine;
    public float totalFlexGrowFactors;
    public float totalFlexShrinkScaledFactors;
    public int endOfLineIndex;
    public final @NotNull ArrayList<YGNode> relativeChildren = new ArrayList<>();
    public float remainingFreeSpace;
    public float mainDim;
    public float crossDim;
}

File: redwood-yoga/src/commonMain/kotlin/app/cash/redwood/yoga/YGCollectFlexItemsRowValues.kt
package app.cash.redwood.yoga

/** Type originates from: Utils.h */
class YGCollectFlexItemsRowValues {
  var itemsOnLine = 0
  var sizeConsumedOnCurrentLine = 0f
  var totalFlexGrowFactors = 0f
  var totalFlexShrinkScaledFactors = 0f
  var endOfLineIndex = 0
  val relativeChildren = ArrayList<YGNode>()
  var remainingFreeSpace = 0f
  var mainDim = 0f
  var crossDim = 0f
}
